lib/jobs/message_latency_report.ex

```diff
@@ -123,15 +135,39 @@ defmodule Jobs.MessageLatencyReport do
         station: station
       ]
     end)
-    |> Enum.sort_by(fn row -> row[:seconds] end)
-    |> Enum.with_index()
+  end
+
+  defp filter_by_date(row, date) do
+    # Get the day of month (0-31) from the date for which the logic is running
+    [_, _, current_day] = date |> String.split("-")
```

Comment:
It might look a little cleaner to convert `date` to a `Date` right off the bat and access the day that way. You are already converting it a couple of lines down so shouldn't throw too much off. This works though so not a blocking suggestion.

Reply:
Oh good point! I like that suggestion and just tweaked to code. Cleaner indeed!